### PR TITLE
Short channel id check

### PR DESCRIPTION
**Pull request by rustyrussell**
With some drive-by fixes along the way...

### CODE REVIEW

**Comment by rustyrussell** [comment]
Indeed, we should handle conflicts generally better (we're supposed to blacklist).  But we should be able to short-circuit the short-id check if the channel is already known, at least.  And will address other fixes once I'm back...

File: channeld/channel.c [hunk 1]
```diff
@@ -372,6 +375,11 @@ static u8 *create_channel_update(const tal_t *ctx,
 	u16 flags;
 	u8 *cupdate, *msg;
 
+	/* Identical timestamps will be ignored. */
```
[the remark this answers is not in the document]
Comment (rustyrussell) [reply channeld/channel.c]: Oops, fixed.

File: gossipd/routing.c [hunk 2]
```diff
@@ -26,6 +27,26 @@
...
 #define MAX_PROPORTIONAL_FEE (1 << 24)
 
+/* We've unpacked and checked its signatures, now we wait for master to tell
+ * us the txout to check */
+struct pending_cannouncement {
+	struct list_node list;
+
+	/* Unpacked fields here */
+	struct short_channel_id short_channel_id;
+	secp256k1_ecdsa_signature node_signature_1;
```
[the remark this answers is not in the document]
Comment (rustyrussell) [reply gossipd/routing.c]: Indeed, I returned them to being local vars.

**Comment by rustyrussell** [comment]
OK, I didn't fix the multiple lookups: we should really do that as part of the FIXME which handles conflicting announcements (which means adding a blacklist).

**Comment by cdecker** [comment]
Rebased, will merge once CI passes.

**Pull request merged**